rbac/management/workspace/view.py

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for Workspace management."""
+from django.utils.translation import gettext as _
+from django_filters import rest_framework as filters
+from management.permissions import WorkspaceAccessPermission
+from rest_framework import mixins, serializers, viewsets
+from rest_framework.filters import OrderingFilter
+
+from .model import Workspace
+from .serializer import WorkspaceSerializer
+
+
+class WorkspaceViewSet(
+    mixins.CreateModelMixin,
+    mixins.DestroyModelMixin,
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Workspace View.
+
+    A viewset that provides default `create()`, `destroy`, `retrieve()`,
+    and `list()` actions.
+
+    """
+
+    permission_classes = (WorkspaceAccessPermission,)
```

Comment:
We should leverage Kessel itself for access controls on these APIs (essentially, Kessel-on-Kessel).

I don't have a complete schema for this yet, but I would consider this placeholder until we can do that. It would be a good way to exercise the Relations infrastructure ourselves by using it for RBAC v2 itself.

Reply:
(I guess we need ticket for it - at least from PRBAC side)

Reply:
@lpichler curious if we have a ticket for this now or if we still need to create one?

Reply:
@coderbydesign I created ticket [here](https://issues.redhat.com/browse/RHCLOUD-33934). I put it into same epic but I am not sure whether it is good place. At this moment I created ticket as SPIKE.(we can change once details around schema will be clear )

Thanks for note!

Reply:
That makes sense, thanks for logging this!